concgroup/concgroup.go

```diff
@@ -24,25 +24,25 @@ type ConcGroup struct {
 //
 // Any subsequent call to the Go method will block until it can add an active
 // goroutine without exceeding the configured limit.
-func WithOptions(ctx context.Context, limit int, comp func(identifier string, data interface{}, err error)) (*ConcGroup, context.Context) {
+func WithOptions[T any](ctx context.Context, limit int, comp func(identifier string, data T, err error)) *ConcGroup[T] {
```

Comment:
I also removed the return value of the `context.Context` here since we weren't returning a modified version, so it felt like that should be safe?

I am curious about the intended use of`concgroup` too. Do we want a single context for the entire group or do we imagine there are scenarios where we might want a different context/timeout value per `Go` calls? (e.g. different timeout values). That's not really urgent to answer in this PR but was just something I was curious about when I started reading the code.

Reply:
Those are good questions.  haven't decided so there was a todo somewhere to think about cancel, timeouts etc.  I took it here in the constructor to start because that's what errorgroup did as I was sketching out.

Reply:
Totally, that makes sense to me 👍